arraylist/src/main/java/ua/hillel/ArrayList.java

```diff
@@ -1,68 +1,123 @@
 package ua.hillel;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 /**
  * Implement a list that is based on array. This resizable data structure
  * is based on an array and is a simplified version of {@link java.util.ArrayList}.
  */
 public class ArrayList<T> implements List<T> {
+    private Object[] arrayElements;
+    private int size;
+
+    public ArrayList() {
+        this.arrayElements = new Object[3];
+        this.size = 0;
+    }
+
+    private void createNewArrIfNeed() {
+        if (arrayElements.length == size) {
+            int newCapacity = arrayElements.length * 2;
+            arrayElements = Arrays.copyOf(arrayElements, newCapacity);
+        }
+    }
 
     @Override
     public void add(T element) {
-
+        createNewArrIfNeed();
+        arrayElements[size] = element;
+        size++;
     }
 
     @Override
     public void add(int index, T element) {
-
+        Objects.checkIndex(index, size);
+        createNewArrIfNeed();
+        System.arraycopy(arrayElements, index, arrayElements, index + 1, size - index);
+        arrayElements[index] = element;
+        size++;
     }
 
     @Override
     public T get(int index) {
+        Objects.checkIndex(index, size);
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                return (T) arrayElements[i];
+            }
+        }
```

Comment:
Why do you need a loop for getting an element by index from an array? Why can't you just do like `return (T) arrayElements[i];`?